View matching (#1571)
ibexa/documentation-developer: docs/templating/templates/template_configuration.md

    Identifier\Section: news
```

In the example above, content which is either an article or a blog post is matched,
but it must be in the "News" Section.

### Matching every Content item

When you use no matcher in a rule, this rule always match. Several values are available to declare no matcher:

``` yaml
match: ~
match: true
match: []
```

Such rules can be found in the [default template configuration](https://github.com/ibexa/core/blob/4.5/src/bundle/Core/Resources/config/default_settings.yml#L47).

!!! tip

>     For example, you can prevent every Content item not having a dedicated template to be seen in `full` view but sent to a custom controller.

Reviewer: ```suggestion
    For example, you can ensure that any Content item lacking a dedicated template isn't displayed in `full` view but is instead sent to a custom controller.
```

    ```yaml
    site_group:
        content_view:
            full:
                # Rules for Content Types and specific Content items meant to be displayed in full view:
                # …
                # Rule for other Content items not meant to be displayed in full view:
                no_full_view:

>                     controller: App\Controller\ViewController:noFullViewAction

Reviewer 2: ```suggestion
                    controller: App\Controller\ViewController::noFullViewAction
```

Syntax with one `:` is deprecated in SF

                    template: '@ibexadesign/full/no_full_view.html.twig'
                    match: ~
    ```

>     Still for example, this custom controller could set the response status code as 404 (`$view->setResponse((new Response())->setStatusCode(404));`) and fetch some reverse relations to make suggestions in this error page.

Reviewer: ```suggestion
    This custom controller can also set the response status code to 404 using the following code: `$view->setResponse((new Response())->setStatusCode(404));`, and fetch reverse relations to provide suggestions on the error page.
```
